sno/working_copy/gpkg.py

```diff
@@ -438,119 +398,60 @@ def _drop_spatial_index(self, dbcur, dataset):
 
     @contextlib.contextmanager
     def _suspend_triggers(self, dbcur, dataset):
         self._drop_triggers(dbcur, dataset)
-        try:
-            yield
-        finally:
-            self._create_triggers(dbcur, dataset)
-
-    def update_gpkg_contents(self, dataset, change_time):
-        table_name = dataset.table_name
-
-        with self.session() as db:
-            if dataset.has_geometry:
-                geom_col = dataset.geom_column_name
-                sql = f"""
-                    UPDATE gpkg_contents
-                    SET
-                        min_x=(SELECT ST_MinX(Extent({gpkg.ident(geom_col)})) FROM {gpkg.ident(table_name)}),
-                        min_y=(SELECT ST_MinY(Extent({gpkg.ident(geom_col)})) FROM {gpkg.ident(table_name)}),
-                        max_x=(SELECT ST_MaxX(Extent({gpkg.ident(geom_col)})) FROM {gpkg.ident(table_name)}),
-                        max_y=(SELECT ST_MaxY(Extent({gpkg.ident(geom_col)})) FROM {gpkg.ident(table_name)}),
-                        last_change=:last_change
-                    WHERE
-                        table_name=:table_name
-                """
-            else:
-                sql = """
-                    UPDATE gpkg_contents
-                    SET min_x=NULL, min_y=NULL, max_x=NULL, max_y=NULL,
-                        last_change=:last_change
-                    WHERE
-                        table_name=:table_name
-                """
-
-            rc = db.execute(
-                sql,
-                {
-                    "last_change": change_time.strftime(
-                        "%Y-%m-%dT%H:%M:%S.%fZ"
-                    ),  # GPKG Spec Req.15
-                    "table_name": table_name,
-                },
-            ).rowcount
-            assert rc == 1, f"gpkg_contents update: expected 1Δ, got {rc}"
-
-    def get_db_tree(self, table_name="*"):
-        with self.session() as db:
-            r = db.execute(
-                f"""
-                    SELECT value
-                    FROM {self.STATE_TABLE}
-                    WHERE table_name=:table_name AND key='tree';
-                """,
-                {"table_name": table_name},
-            )
-            row = r.fetchone()
-            if not row:
-                # It's okay to not have anything in the tree table - it might just mean there are no commits yet.
-                # It might also mean that the working copy is not yet initialised - see WorkingCopy.get
-                return None
-
-            wc_tree_id = row[0]
-            return wc_tree_id
+        yield
+        self._create_triggers(dbcur, dataset)
 
     def _create_triggers(self, db, dataset):
-        table = dataset.table_name
-        pkf = gpkg.ident(dataset.primary_key)
-        ts = gpkg.param_str(table)
+        table_identifier = self.table_identifier(dataset)
+        pk_column = self.quote(dataset.primary_key)
+
+        # SQLite doesn't let you do param substitutions in CREATE TRIGGER:
```

Comment:
Wow: https://stackoverflow.com/questions/29032750/using-sqlite-trigger-with-argument

that's madness

Reply:
Ack